raster/src/main/scala/geotrellis/raster/rasterize/Rasterizer.scala

```diff
@@ -205,6 +221,29 @@ object Rasterizer {
     * LineString.  The iteration happens in the direction from the
     * first point to the last point.
     */
+  def foreachCellByLineString(
+    line: Line,
+    re: RasterExtent,
+    options: Options
+  )(f: (Int, Int) => Unit) {
+    val cells = (for(coord <- line.jtsGeom.getCoordinates()) yield {
+      (re.mapXToGrid(coord.x), re.mapYToGrid(coord.y))
+    }).toList
```

Comment:
Rasterizer is a performance sensitive part of code in my experience. `getCoordinates` already returns an `Array[Coordinate]` this loop and loop below can be rolled into one. That avoids an extra collection allocation and the takes care of indexing cost comment below.